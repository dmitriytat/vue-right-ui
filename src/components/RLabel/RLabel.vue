<template>
  <label
    :for="forId"
    :class="b({ active, focused, invalid })"
  >
    <slot />
  </label>
</template>

<script>
export default {
  name: 'RLabel',
  block: 'r-label',

  props: {
    /**
     * Id of input element
     */
    forId: {
      type: String,
      required: true,
    },

    /**
     * Is field focused
     */
    active: {
      type: Boolean,
      default: false,
    },

    /**
     * Is field focused
     */
    focused: {
      type: Boolean,
      default: false,
    },

    /**
     * Is field invalid
     */
    invalid: {
      type: Boolean,
      default: false,
    },
  },
};
</script>

<style>
  .r-label {
    position: relative;
    z-index: 10;
    display: block;
    height: 24px;
    color: var(--color-gray);
    font: normal 17px/24px var(--font-family);
    transform: translateY(29px);
    transition: transform var(--transition-base), font-size var(--transition-base), color var(--transition-base);
    pointer-events: none;

    &_active {
      font-size: 14px;
      transform: translateY(0);
      pointer-events: auto;
    }

    &_focused {
      color: var(--color-blue);
    }

    &_invalid {
      color: var(--color-red);
    }
  }
</style>

<docs>
  Component for labeling input in field

  ```
  <RLabel for-id="#" active>Label</RLabel>
  ```
</docs>
