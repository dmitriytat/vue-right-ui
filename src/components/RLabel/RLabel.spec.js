import { shallowMount } from '@vue/test-utils';
import { createRenderer } from 'vue-server-renderer';

import RLabel from './RLabel.vue';

describe('RLabel', () => {
  const renderer = createRenderer();

  it('should render label', () => {
    const wrapper = shallowMount(RLabel, {
      propsData: {
        id: 'label',
      },
    });

    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err);

      expect(str).toMatchSnapshot();
    });
  });

  it('should render active label', () => {
    const wrapper = shallowMount(RLabel, {
      propsData: {
        id: 'label',
        active: true,
      },
    });

    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err);

      expect(str).toMatchSnapshot();
    });
  });
});
