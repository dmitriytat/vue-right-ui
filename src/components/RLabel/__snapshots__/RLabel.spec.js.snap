// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RLabel should render active label 1`] = `<label for="input" id="label" class="r-label r-label_active"></label>`;

exports[`RLabel should render label 1`] = `<label for="input" id="label" class="r-label"></label>`;
